includes/payment-methods/class-upe-payment-method.php

```diff
@@ -171,11 +171,26 @@ public function is_enabled_at_checkout( string $account_country ) {
 
 		// This part ensures that when payment limits for the currency declared, those will be respected (e.g. BNPLs).
 		if ( [] !== $this->limits_per_currency ) {
-			$currency = get_woocommerce_currency();
+			$order = null;
+			if ( is_wc_endpoint_url( 'order-pay' ) ) {
+				$order = wc_get_order( absint( get_query_var( 'order-pay' ) ) );
```

Comment:
(minor) WooCommerce Core does a similar check on the currency amount (although we're not leveraging it because of the multi-currency setup): https://github.com/woocommerce/woocommerce/blob/7eabbe4ecd7dd1cff6a36055c8a07c3bdf64abb1/plugins/woocommerce/includes/abstracts/abstract-wc-payment-gateway.php#L290-L308

However, I noticed that they first do:
```php
$order_id = absint( get_query_var( 'order-pay' ) )
```

And then:
```php
if ( 0 < $order_id ) {
  // proceeds with getting the order
}
```

However, we're not doing the check on the `$order_id`, here.
I'm not sure if you considered it - but wanted to raise it anyways. I'm guessing that if we call `wc_get_order()` with `''` (empty string - the default value returned by `get_query_var()`, we might get the order with id `0`. Is that something we should be concerned about?
I don't think we'd need to be concerned about it, but wanted to double-check with you.

Reply:
I followed the same approach we already use to retrieve the order in the Pay for Order flow: https://github.com/Automattic/woocommerce-payments/blob/c061a0874a4af2d3dbe25a88c6cca44015d02a5a/includes/class-wc-payments-checkout.php#L264-L268

I also checked the source code for `wc_get_order` and it should return false in case we sent either `0` or `''`.
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-order-factory.php#L28-L32
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-order-factory.php#L210-L221